### PR TITLE
Dropping Python 3.7

### PR DESCRIPTION
**Pull request by mzbroch**
- Drops Python 3.7
- Updates Pylint to newer version

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -119,14 +119,14 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (glennmatthews) [inline .github/workflows/ci.yml:122]: Does it make sense to add 3.11 at this time as well?

File: nautobot_bgp_models/migrations/0002_viewsets_migration.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=missing-module-docstring,missing-function-docstring,missing-class-docstring,invalid-name
```
Comment (glennmatthews) [inline nautobot_bgp_models/migrations/0002_viewsets_migration.py:1]: I'm surprised on the need for this - I don't think I've ever seen a migrations file need pylint disables as I think `pylint-django` silences most of them automatically?

**Review by glennmatthews — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mzbroch) [reply .github/workflows/ci.yml]: Added

File: nautobot_bgp_models/migrations/0002_viewsets_migration.py (hunk 2 above)
Comment (mzbroch) [reply nautobot_bgp_models/migrations/0002_viewsets_migration.py:1]: Will need to take a closer look about exact reason of that. 👍

**Pull request merged**